engine/src/main/java/org/terasology/input/MenuInputSystem.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.input;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.config.Config;
+import org.terasology.engine.Time;
+import org.terasology.engine.module.ModuleManager;
+import org.terasology.engine.subsystem.DisplayDevice;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.input.device.MouseAction;
+import org.terasology.input.device.MouseDevice;
+import org.terasology.input.device.nulldevices.NullMouseDevice;
+import org.terasology.input.events.InputEvent;
+import org.terasology.input.events.LeftMouseDownButtonEvent;
+import org.terasology.input.events.LeftMouseUpButtonEvent;
+import org.terasology.input.events.MouseAxisEvent;
+import org.terasology.input.events.MouseButtonEvent;
+import org.terasology.input.events.MouseDownButtonEvent;
+import org.terasology.input.events.MouseUpButtonEvent;
+import org.terasology.input.events.MouseWheelEvent;
+import org.terasology.input.events.MouseXAxisEvent;
+import org.terasology.input.events.MouseYAxisEvent;
+import org.terasology.input.events.RightMouseDownButtonEvent;
+import org.terasology.input.events.RightMouseUpButtonEvent;
+import org.terasology.input.internal.BindableButtonImpl;
```

Comment:
My bad! I forgot to remove those.
The difficult part now is to get the keyboard input on the menu. The key bindings class is slightly (quite) dependent on the original input system class. I was playing around with that, hence all the imports. 
Will remove.